docs/references/sdk/philosophy.mdx

```diff
@@ -0,0 +1,40 @@
+---
+description: The guiding principle for building a Clerk SDK.
+---
+
+# Philosophy
+
+A reason why there’s not only a React SDK for Clerk but also a Next.js, Remix, and Astro one is that they’ll deliver a better integration than a general purpose SDK. Each framework has slightly different APIs, capabilities, and conventions.
+
+> [!IMPORTANT]
+> **Optimize for the user’s developer experience.** These principles, conventions, and instructions below are not strict rules. Use your best judgement to adjust as needed. Adhere to what makes most sense for the programming language or framework.
+
+Here’s a non-exhaustive list of things to take into consideration to deliver a great developer experience.
+
+## Performance
+
+Developers and marketers alike care about their website's [performance scoring](https://developer.chrome.com/docs/lighthouse/performance/performance-scoring). Ensure efficient loading of your SDK and reduce bundle size & dependencies. Less dependencies also mean less maintenance and security concerns.
+
+## Enable best practices
+
+For your SDK, follow the framework's or language's best practices and consider using their specialized patterns over generalized ones (e.g. special script loading). For your users, provide APIs to leverage those best practices.
+
+## Documentation
+
+Spend time writing an exhaustive, yet organized `README` that enables your users to get started with your SDK. If you want to go above and beyond, create your own documentation site (recommended: [Starlight](https://starlight.astro.build/)). Provide examples on how to use your SDK, e.g. with a separate example repository.
+
+## Convenience
+
+Make SDK usage convenient. If the SDK can handle something under the hood which otherwise would require an action by the user, do it. Clerk's components are its API as that abstraction is powerful and convenient.
```

Comment:
"Clerk's components are its API" doesn't make sense - what were you trying to say here?

Reply:
You might interact/use a product through a REST API or through an SDK that exposes certain low-level methods. That's the API surface that they expose to their users.

Clerk's API are the prebuilt React components. Just place them in your React application and you're good. It's super convenient.

That's what I wanted to convey 😊